test/unit/test_fvm_layout.cpp

```diff
@@ -129,6 +129,60 @@ namespace {
     }
 } // namespace
 
+TEST(fvm_layout, compatible_mechanisms) {
```

Comment:
Another option for compression:
All checks have the same layout, except the decoration step. You might prefer to pull that out into a lambda
like this for example
```cxx
    // S is not needed except for my laziness to look up the proper type.
    template<typename P, typename S> check_failure(S& system, P painter) {
        auto system = two_cell_system();

        painter(system);

        auto cells = system.cells();
        check_two_cell_system(cells);
        fvm_cv_discretization D = fvm_cv_discretize(cells, gprop.default_parameters);

        EXPECT_THROW(fvm_build_mechanism_data(gprop, cells, D), arb::cable_cell_error);
    }
```
and write this 
```cxx
do_check(system, [](auto& sys) {system.descriptions[0].decorations.place(system.builders[0].location({1, 0.4}), "hh", "syn0"); });
```
making the code more compact and saves some duplication, but uses more complex features.
Your call on whether you like it or not.

Reply:
Okay, I like it.